api-reference/v2/api-reference-template.md

```diff
@@ -0,0 +1,188 @@
+# API reference overview
+
+{Before using this template, read [the accompanying guide](api-reference-guide.md) to this template}.
+
+## Introduction
+
+The {product name} APIs are REST APIs that provides {xx features}.
+
+## Base URL
+
+The APIs referenced in this documentation use the following base URL:
+
+```
+https://api.example.com
+```
+
+
+## Authorization
+
+All requests to the API must be authenticated and authorized before they can access or manage resources on the {environment or platform}.
+
+The {product name} API uses the **{auth_type}** authorization type.
+
+You can authenticate your requests by {using authentication type and method}.
+
+For example:
+
+```
+<EXAMPLE>
+```
+
+
+API requests without valid authentication will fail with a status of {error code}.
+
+
+## Versioning
+
+{This section is optional.}
+
+The {product name} APIs provide {x} versions of API endpoints...
+
+## Pagination
+
+{This section is optional.}
+
+You can use {xx} to paginate the results returned by the {product name} APIs. If not specified, the default values are ...
+
+## Rate limiting and throttling
+
+{This section is optional.}
+
+The {product name} APIs use a {xx} rate limiting strategy. The maximum number of requests allowed to access a {resource/endpoint/..} is {xx} requests per {time period}.
+
+If the number of requests exceeds that limit, an HTTP `429 Too Many Request` error is returned.
+
+## Status code
+
+The {product name} APIs use standard HTTP response codes.
+
+| Status code | Message           | Description   |
+|-------------|-------------------|---------------|
+| 200 `OK`    | Request succeeds. | {description} |
+|             |                   |               |
+|             |                   |               |
+
+# {Resource name}
+
+The {resource name} is used to {xxx}.
+
+## Data model
+
+| Attribute | Type   | Required/Optional | Description                |
+|-----------|--------|-------------------|----------------------------|
+| {id}      | string | REQUIRED          | {Unique identifier of xx.} |
+| {name}    | string | Optional          | {Name of xx}.              |
+|           |        |                   |                            |
+
+
+## Example
+
+```
+<EXAMPLE>
+```
+
+## Endpoints
+
+Use the following endpoints to interact with the {resource name} entities.
+
+| Method | Endpoint name                           | Description           |
+|--------|-----------------------------------------|-----------------------|
+| POST   | {[Endpoint name A](link_to_endpoint_a)} | Creates a {resource}. |
+| GET    | {[Endpoint name B](link_to_endpoint_b)} | Returns a {resource}. |
+|        |                                         |                       |
+
+
+# {Endpoint name}
+
+{Provide a one-line description of what the API does. Starts with a verb.}
```

Comment:
Suggestion: We probably shouldn't assume the reader (a software developer) has a good understanding of what a "verb" is, so this description should be followed by an example.